optimum/exporters/openvino/__main__.py

```diff
@@ -247,6 +247,45 @@ class StoreAttr(object):
 
         GPTQQuantizer.post_init_model = post_init_model
 
+    # Apply quantization in hybrid mode to Stable Diffusion before export
+    if (
+        library_name == "diffusers"
+        and ov_config
+        and ov_config.quantization_config
+        and ov_config.quantization_config.get("dataset", None)
+    ):
+        import huggingface_hub
+
+        model_info = huggingface_hub.model_info(model_name_or_path, revision=revision)
```

Comment:
what if it is a local directory instead of published on hub model? Will this work?

Reply:
No this will not work for model hosted locally, could we load the model's config instead https://huggingface.co/stabilityai/stable-diffusion-xl-base-1.0/blob/main/model_index.json ?

Reply:
I can use `model.__class__.__name__`, where `model = TasksManager.get_model_from_task(..)`

Reply:
Something that could be used to infer the task given a model id of a model hosted on the hub is to use  `TasksManager.infer_task_from_model` https://github.com/huggingface/optimum/blob/dedb852eaea3d3d899920f4017b9fd899eb2756b/optimum/exporters/tasks.py#L1601. Like this we can move it here as https://github.com/huggingface/optimum-intel/blob/main/optimum/commands/export/openvino.py#L191. For model hosted locally the task needs to be provided https://github.com/huggingface/optimum/blob/dedb852eaea3d3d899920f4017b9fd899eb2756b/optimum/exporters/tasks.py#L1552 (already true for model exported with a CLI without any quantization)

Reply:
what about moving conversion of sd from export_main into separated function  and calling quantization after conversion when models saved on  disk happens if config provided?

similar way used for weights compression, but weights compression applied on single model and we do not need to have whole pipeline, that is why I think it is better to separate sd comodels conversion to know where pipeline conversion finished